velox/connectors/hive/storage_adapters/s3fs/tests/S3InsertTest.cpp

```diff
@@ -14,18 +14,17 @@
  * limitations under the License.
  */
 
-#include <folly/init/Init.h>
 
-#include "gtest/gtest.h"
-#include "velox/common/file/FileSystems.h"
 #include "velox/connectors/hive/storage_adapters/s3fs/RegisterS3FileSystem.h"
 #include "velox/connectors/hive/storage_adapters/s3fs/tests/MinioServer.h"
-#include "velox/dwio/parquet/reader/ParquetReader.h"
 #include "velox/exec/TableWriter.h"
 #include "velox/exec/tests/utils/AssertQueryBuilder.h"
 #include "velox/exec/tests/utils/HiveConnectorTestBase.h"
 #include "velox/exec/tests/utils/PlanBuilder.h"
 
+#include <folly/init/Init.h>
```

Comment:
these standard includes should go in the top, as per our coding style document.

Reply:
@pedroerp Thanks for reviewing, I've been thinking to discuss the include order with you for a long time but didn't have time. I read many discussions about include orders online, e.g. https://stackoverflow.com/questions/2762568/c-c-include-header-file-order https://blog.knatten.org/2010/07/01/the-order-of-include-directives-matter/ and the majority agreement is to have local to global order:

1. The prototype/interface header for this implementation (ie, the .h/.hh file that corresponds to this .cpp/.cc file).
2. Other headers from the same project, as needed.
3. Headers from other non-standard, non-system libraries (for example, Qt, Eigen, etc).
4. Headers from other "almost-standard" libraries (for example, Boost)
5. Standard C++ headers (for example, iostream, functional, etc.)
6. Standard C headers (for example, cstdint, dirent.h, etc.)

The main reason is to avoid hidden dependencies.  E.g. if myclass.cpp includes <string> then "myclass.h", there's no way to catch at build time that myclass.h may itself depend on string; so if later onsomeone includes myclass.h but don't need string, he will get an error that needs to be fixed either in the cpp or in the header itself. This is well explained in https://blog.knatten.org/2010/07/01/the-order-of-include-directives-matter/. 

As someone mentioned in the StackOverflow discussion, It was also mentioned in "Thinking in C++" referencing Lakos' "Large Scale C++ Software Design". Also [Google C++ Style Guide](http://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes) recommends using this order.

I think these reasoning makes a lot of sense, so would suggest we follow the local->global order and update the coding guideline. I can start a discussion if you think it's necessary.

Reply:
@pedroerp I started discussion https://github.com/facebookincubator/velox/discussions/7723